### PR TITLE
Add Stroke/Border to LibTxt and dart:ui

### PR DESCRIPTION
**Pull request by GaryQian**
This adds the border/stroke feature requested in https://github.com/flutter/flutter/issues/24108

Three properties are added in TextStyle:

- StrokeStyle
  - under
  - over
- StrokeColor
- StrokeWidth

These properties allow LibTxt to natively draw stroke and borders.

### CODE REVIEW

**Comment by GaryQian** [comment]
cc @Rusino

File: lib/ui/text.dart [hunk 1]
```diff
@@ -431,6 +431,23 @@ enum TextDecorationStyle {
   wavy
 }
 
+
+/// The type of Stroke to draw.
+enum StrokeStyle {
+  /// No stroke.
+  kNone,
```
Comment (jason-simmons) [inline lib/ui/text.dart]: remove the "k" prefix from these values

**Comment by GaryQian** [comment]
Thanks for bringing that up. My intention with this change is to actually add functionality that I did not see was directly possible with just a Paint. Primarily, the stroke over fill and fill over stroke capability. This is lightly inspired by the level of control available in Adobe after effects (see https://helpx.adobe.com/after-effects/using/formatting-characters-character-panel.html), which I have (personally) found to be very powerful.

The question is whether this functionality should be added, and if so, how to allow it to be added logically in conjunction with the foreground property.

**Comment by dnfield** [comment]
I would be in favor of not having libtxt or the engine handling multiple draws of text.  As written, setting these properties will involve double painting the text, whereas setting the foreground paint properties will result in a single paint (but will not allow you to independently set stroke and fill).

Drawing stroked and filled text is expensive.  We shouldn't make expensive things easy.

**Comment by Hixie** [comment]
This seems redundant with the foreground Paint? I don't understand what this enables.

**Comment by GaryQian** [comment]
I will make this a docs change instead to clarify how to accomplish this.

**Pull request closed without merging**